kernel/sched.c

```diff
@@ -1608,7 +1608,7 @@ static int cpu_mask_mod(k_tid_t thread, uint32_t enable_mask, uint32_t disable_m
 		}
 	}
 
-#if defined(CONFIG_ASSERT) && defined(CONFIG_SCHED_CPU_MASK_PIN_ONLY)
+#ifdef CONFIG_SCHED_CPU_MASK_PIN_ONLY
```

Comment:
Oh.... sorry, I didn't read far enough.  That #ifdef isn't protecting the use of __ASSERT, it's protecting the "m" value which would otherwise be flagged as an unused variable by the compiler.

Reply:
yep, fair enough. Will just remove this commit from this pr.